### PR TITLE
[BEAM-2621] BeamSqlRecordType -> BeamSqlRowType

### PR DESCRIPTION
**Pull request by xumingming**
@XuMingmin @jbonofre @takidau

### CODE REVIEW

File: dsls/sql/src/main/java/org/apache/beam/dsls/sql/BeamSqlEnv.java [hunk 1]
```diff
@@ -84,13 +84,13 @@ public BaseBeamTable findTable(String tableName){
...
-    public BeamCalciteTable(BeamSqlRecordType beamSqlRecordType) {
-      this.beamSqlRecordType = beamSqlRecordType;
+    private BeamSqlRowType beamSqlRowType;
+    public BeamCalciteTable(BeamSqlRowType beamSqlRowType) {
+      this.beamSqlRowType = beamSqlRowType;
     }
     @Override
     public RelDataType getRowType(RelDataTypeFactory typeFactory) {
-      return CalciteUtils.toCalciteRecordType(this.beamSqlRecordType)
+      return CalciteUtils.toCalciteRecordType(this.beamSqlRowType)
```
Comment (XuMingmin) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/BeamSqlEnv.java]: to `return CalciteUtils.toCalciteRowType(this.beamSqlRowType)`

File: dsls/sql/src/main/java/org/apache/beam/dsls/sql/rel/BeamAggregationRel.java [hunk 2]
```diff
@@ -105,22 +105,22 @@ public BeamAggregationRel(RelOptCluster cluster, RelTraitSet traits
...
     return mergedStream;
   }
 
   /**
    * Type of sub-rowrecord used as Group-By keys.
    */
-  private BeamSqlRecordType exKeyFieldsSchema(RelDataType relDataType) {
-    BeamSqlRecordType inputRecordType = CalciteUtils.toBeamRecordType(relDataType);
+  private BeamSqlRowType exKeyFieldsSchema(RelDataType relDataType) {
+    BeamSqlRowType inputRecordType = CalciteUtils.toBeamRowType(relDataType);
```
Comment (XuMingmin) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/rel/BeamAggregationRel.java]: BeamSqlRowType input**Row**Type = CalciteUtils.toBeamRowType(relDataType);

File: dsls/sql/src/main/java/org/apache/beam/dsls/sql/schema/BaseBeamTable.java [hunk 3]
```diff
@@ -23,12 +23,12 @@
...
-  public BaseBeamTable(BeamSqlRecordType beamSqlRecordType) {
-    this.beamSqlRecordType = beamSqlRecordType;
+  protected BeamSqlRowType beamSqlRowType;
+  public BaseBeamTable(BeamSqlRowType beamSqlRowType) {
+    this.beamSqlRowType = beamSqlRowType;
   }
 
-  @Override public BeamSqlRecordType getRecordType() {
-    return beamSqlRecordType;
+  @Override public BeamSqlRowType getRecordType() {
```
Comment (XuMingmin) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/schema/BaseBeamTable.java]: @Override public BeamSqlRowType get**Row**Type() {

File: dsls/sql/src/main/java/org/apache/beam/dsls/sql/schema/BeamSqlTable.java [hunk 4]
```diff
@@ -48,5 +48,5 @@
   /**
    * Get the schema info of the table.
    */
-   BeamSqlRecordType getRecordType();
+   BeamSqlRowType getRecordType();
```
Comment (XuMingmin) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/schema/BeamSqlTable.java]: BeamSqlRowType get**Row**Type();

File: dsls/sql/src/main/java/org/apache/beam/dsls/sql/schema/kafka/BeamKafkaCSVTable.java [hunk 5]
```diff
@@ -66,9 +65,9 @@ public BeamKafkaCSVTable(BeamSqlRecordType beamSqlRecordType, String bootstrapSe
    */
   public static class CsvRecorderDecoder
       extends PTransform<PCollection<KV<byte[], byte[]>>, PCollection<BeamSqlRow>> {
-    private BeamSqlRecordType recordType;
+    private BeamSqlRowType recordType;
```
Comment (XuMingmin) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/schema/kafka/BeamKafkaCSVTable.java]: private BeamSqlRowType **row**Type;

File: dsls/sql/src/main/java/org/apache/beam/dsls/sql/schema/kafka/BeamKafkaCSVTable.java [hunk 6]
```diff
@@ -91,9 +90,9 @@ public void processElement(ProcessContext c) {
    */
   public static class CsvRecorderEncoder
       extends PTransform<PCollection<BeamSqlRow>, PCollection<KV<byte[], byte[]>>> {
-    private BeamSqlRecordType recordType;
+    private BeamSqlRowType recordType;
```
Comment (XuMingmin) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/schema/kafka/BeamKafkaCSVTable.java]: private BeamSqlRowType **row**Type;

File: dsls/sql/src/main/java/org/apache/beam/dsls/sql/schema/kafka/BeamKafkaCSVTable.java [hunk 7]
```diff
@@ -91,9 +90,9 @@ public void processElement(ProcessContext c) {
...
     private CSVFormat format;
-    public CsvRecorderEncoder(BeamSqlRecordType recordType, CSVFormat format) {
+    public CsvRecorderEncoder(BeamSqlRowType recordType, CSVFormat format) {
```
Comment (XuMingmin) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/schema/kafka/BeamKafkaCSVTable.java]: BeamSqlRowType **row**Type

File: dsls/sql/src/main/java/org/apache/beam/dsls/sql/transform/BeamAggregationTransforms.java [hunk 8]
```diff
@@ -57,11 +57,11 @@
    * Merge KV to single record.
    */
   public static class MergeAggregationRecord extends DoFn<KV<BeamSqlRow, BeamSqlRow>, BeamSqlRow> {
-    private BeamSqlRecordType outRecordType;
+    private BeamSqlRowType outRecordType;
```
Comment (XuMingmin) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/transform/BeamAggregationTransforms.java]: private BeamSqlRowType out**Row**Type;

File: dsls/sql/src/main/java/org/apache/beam/dsls/sql/transform/BeamAggregationTransforms.java [hunk 9]
```diff
@@ -206,7 +206,7 @@ public AggregationAdaptor(List<AggregateCall> aggregationCalls,
           break;
         }
       }
-      finalRecordType = BeamSqlRecordType.create(outFieldsName, outFieldsType);
+      finalRecordType = BeamSqlRowType.create(outFieldsName, outFieldsType);
```
Comment (XuMingmin) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/transform/BeamAggregationTransforms.java]: final**Row**Type

File: dsls/sql/src/main/java/org/apache/beam/dsls/sql/transform/BeamSqlProjectFn.java [hunk 10]
```diff
@@ -34,10 +34,10 @@
 public class BeamSqlProjectFn extends DoFn<BeamSqlRow, BeamSqlRow> {
   private String stepName;
   private BeamSqlExpressionExecutor executor;
-  private BeamSqlRecordType outputRecordType;
+  private BeamSqlRowType outputRecordType;
```
Comment (XuMingmin) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/transform/BeamSqlProjectFn.java]: private BeamSqlRowType output**Row**Type;

File: dsls/sql/src/main/java/org/apache/beam/dsls/sql/utils/CalciteUtils.java [hunk 11]
```diff
@@ -76,27 +76,27 @@ public static Integer toJavaType(SqlTypeName typeName) {
...
     }
-    return BeamSqlRecordType.create(fieldNames, fieldTypes);
+    return BeamSqlRowType.create(fieldNames, fieldTypes);
   }
 
   /**
    * Create an instance of {@code RelDataType} so it can be used to create a table.
    */
-  public static RelProtoDataType toCalciteRecordType(final BeamSqlRecordType that) {
+  public static RelProtoDataType toCalciteRecordType(final BeamSqlRowType that) {
```
Comment (XuMingmin) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/utils/CalciteUtils.java]: public static RelProtoDataType toCalcite**Row**Type(final BeamSqlRowType that) {

File: dsls/sql/src/test/java/org/apache/beam/dsls/sql/BeamSqlDslBase.java [hunk 12]
```diff
@@ -52,7 +52,7 @@
   @Rule
   public ExpectedException exceptions = ExpectedException.none();
 
-  public static BeamSqlRecordType recordTypeInTableA;
+  public static BeamSqlRowType recordTypeInTableA;
```
Comment (XuMingmin) [inline dsls/sql/src/test/java/org/apache/beam/dsls/sql/BeamSqlDslBase.java]: public static BeamSqlRowType **row**TypeInTableA;

File: dsls/sql/src/test/java/org/apache/beam/dsls/sql/interpreter/BeamSqlFnExecutorTestBase.java [hunk 13]
```diff
@@ -57,7 +57,7 @@
       RelDataTypeSystem.DEFAULT);
   public static RelDataType relDataType;
 
-  public static BeamSqlRecordType beamRecordType;
+  public static BeamSqlRowType beamRecordType;
```
Comment (XuMingmin) [inline dsls/sql/src/test/java/org/apache/beam/dsls/sql/interpreter/BeamSqlFnExecutorTestBase.java]: public static BeamSqlRowType beam**Row**Type;

**Comment by xumingming** [comment]
Good catches, @XuMingmin ！

**Comment by JingsongLi** [comment]
+R @JingsongLi, I'll take a quick look.

**Comment by JingsongLi** [comment]
@xumingming There are still many inputRecord. Should be inputRow?

**Comment by xumingming** [comment]
Rebase done.

**Comment by xumingming** [comment]
Retest this please

**Comment by JingsongLi** [comment]
Retest this please

**Comment by xumingming** [comment]
Retest this please

**Comment by JingsongLi** [comment]
Merged, feel free to close.

**Comment by xumingming** [comment]
Thanks @JingsongLi @XuMingmin !

**Comment by robertwb** [comment]
Weren't we going to drop the "SQL" and just name these BeamRecordType or BeamRowType?

**Pull request closed without merging**